### PR TITLE
Add some new types

### PR DESCRIPTION
**Pull request by gergocs**
new types: JERRY_OBJECT_TYPE_NAMESPACE, JERRY_OBJECT_TYPE_MODULE, JERRY_OBJECT_TYPE_PROMISE, JERRY_OBJECT_TYPE_DATAVIEW, JERRY_OBJECT_TYPE_SITERATOR, JERRY_OBJECT_TYPE_ERROR, JERRY_OBJECT_TYPE_ARRAYBUFF

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -39,14 +39,20 @@ Enum that contains JerryScript **object** value types:
 
  - JERRY_OBJECT_TYPE_NONE - Non-object type
  - JERRY_OBJECT_TYPE_GENERIC - Generic JavaScript object without any internal property
+ - JERRY_OBJECT_TYPE_NAMESPACE - Namespace object
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:42]: Module namespace object
(I think module is an important info here)

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -39,14 +39,20 @@ Enum that contains JerryScript **object** value types:
...
  - JERRY_OBJECT_TYPE_ARRAY - Array object
  - JERRY_OBJECT_TYPE_PROXY - Proxy object
  - JERRY_OBJECT_TYPE_SCRIPT - Script object (see [jerry_parse](#jerry_parse))
  - JERRY_OBJECT_TYPE_MODULE - Module object (see [jerry_parse](#jerry_parse))
+ - JERRY_OBJECT_TYPE_PROMISE - Promise object
+ - JERRY_OBJECT_TYPE_DATAVIEW - Dataview object
  - JERRY_OBJECT_TYPE_FUNCTION - Function object (see [jerry_function_get_type](#jerry_function_get_type))
  - JERRY_OBJECT_TYPE_TYPEDARRAY - %TypedArray% object (see [jerry_get_typedarray_type](#jerry_get_typedarray_type))
  - JERRY_OBJECT_TYPE_ITERATOR - Iterator object (see [jerry_iterator_get_type](#jerry_get_typedarray_type))
+ - JERRY_OBJECT_TYPE_SITERATOR - String iterator object
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:52]: SITERATOR :)
Iterator is iterator, `jerry_iterator_get_type` should provide more detailed info.

File: docs/02.API-REFERENCE.md [hunk 3]
```diff
@@ -2126,6 +2132,81 @@ jerry_value_is_boolean (const jerry_value_t value)
...
+
+**Summary**
+
+Returns whether the given `jerry_value_t` is a true value.
+
+**Prototype**
+
+```c
+bool
+jerry_value_is_true (const jerry_value_t value);
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:2145]: This should be in a different patch.

**Review by zherczeg — CHANGES_REQUESTED** [review]
Documentation update is missing.

File: docs/02.API-REFERENCE.md [hunk 4]
```diff
@@ -39,14 +39,19 @@ Enum that contains JerryScript **object** value types:
...
  - JERRY_OBJECT_TYPE_SCRIPT - Script object (see [jerry_parse](#jerry_parse))
  - JERRY_OBJECT_TYPE_MODULE - Module object (see [jerry_parse](#jerry_parse))
+ - JERRY_OBJECT_TYPE_PROMISE - Promise object
+ - JERRY_OBJECT_TYPE_DATAVIEW - Dataview object
  - JERRY_OBJECT_TYPE_FUNCTION - Function object (see [jerry_function_get_type](#jerry_function_get_type))
  - JERRY_OBJECT_TYPE_TYPEDARRAY - %TypedArray% object (see [jerry_get_typedarray_type](#jerry_get_typedarray_type))
  - JERRY_OBJECT_TYPE_ITERATOR - Iterator object (see [jerry_iterator_get_type](#jerry_get_typedarray_type))
  - JERRY_OBJECT_TYPE_CONTAINER - Container object (see [jerry_get_container_type](#jerry_get_container_type))
+ - JERRY_OBJECT_TYPE_ERROR - Error object
+ - JERRY_OBJECT_TYPE_ARRAYBUFF - Array buffer object
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:54]: Maybe this could be JERRY_OBJECT_TYPE_ARRAYBUFFER. No need for abbreviation.

**Review by zherczeg — CHANGES_REQUESTED** [review]
You need to add API tests for testing these types.

File: tests/unit-core/test-api-objecttype.c [hunk 5]
```diff
@@ -40,6 +40,16 @@ test_ext_function (const jerry_call_info_t *call_info_p, /**< call information *
   return jerry_create_boolean (true);
 } /* test_ext_function */
 
+static jerry_object_type_t
+test_namespace (const jerry_parse_options_t module_parse_options) /** module options */
+{
+  jerry_value_t module = jerry_parse ((const jerry_char_t *) "", 0, (&module_parse_options));
+  jerry_module_link (module, NULL, NULL);
```
Comment (zherczeg) [inline tests/unit-core/test-api-objecttype.c:47]: `jerry_release_value (jerry_module_link (module, NULL, NULL))`

File: tests/unit-core/test-api-objecttype.c [hunk 6]
```diff
@@ -108,6 +124,8 @@ main (void)
     ENTRY (JERRY_OBJECT_TYPE_FUNCTION, jerry_create_external_function (test_ext_function)),
     ENTRY (JERRY_OBJECT_TYPE_FUNCTION, EVALUATE (getter_function)),
     ENTRY (JERRY_OBJECT_TYPE_FUNCTION, EVALUATE (setter_function)),
+    ENTRY_IF (JERRY_OBJECT_TYPE_ERROR, EVALUATE (error_object), JERRY_FEATURE_ERROR_MESSAGES),
+    ENTRY_IF (JERRY_OBJECT_TYPE_ARRAYBUFFER, jerry_create_arraybuffer (10), JERRY_FEATURE_DATAVIEW),
```
Comment (zherczeg) [inline tests/unit-core/test-api-objecttype.c:128]: jerry_create_arraybuffer depends on dataview?

File: tests/unit-core/test-api-objecttype.c [hunk 7]
```diff
@@ -97,6 +111,8 @@ main (void)
 
     ENTRY (JERRY_OBJECT_TYPE_SCRIPT, PARSE (NULL)),
     ENTRY_IF (JERRY_OBJECT_TYPE_MODULE, PARSE (&module_parse_options), JERRY_FEATURE_MODULE),
+    ENTRY_IF (JERRY_OBJECT_TYPE_PROMISE, jerry_create_promise (), JERRY_FEATURE_PROMISE),
+    ENTRY_IF (JERRY_OBJECT_TYPE_DATAVIEW, jerry_create_dataview(jerry_create_arraybuffer (10), 0, 4), JERRY_FEATURE_DATAVIEW),
```
Comment (zherczeg) [inline tests/unit-core/test-api-objecttype.c:115]: result of jerry_create_arraybuffer is not freed

File: docs/02.API-REFERENCE.md [hunk 8]
```diff
@@ -39,14 +39,19 @@ Enum that contains JerryScript **object** value types:
 
  - JERRY_OBJECT_TYPE_NONE - Non-object type
  - JERRY_OBJECT_TYPE_GENERIC - Generic JavaScript object without any internal property
+ - JERRY_OBJECT_TYPE_NAMESPACE - Module namespace object
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:42]: JERRY_OBJECT_TYPE_MODULE_NAMESPACE

**Review by zherczeg — CHANGES_REQUESTED** [review]

File: tests/unit-core/test-api-objecttype.c [hunk 9]
```diff
@@ -40,6 +40,17 @@ test_ext_function (const jerry_call_info_t *call_info_p, /**< call information *
   return jerry_create_boolean (true);
 } /* test_ext_function */
 
+static jerry_object_type_t
+test_namespace (const jerry_parse_options_t module_parse_options) /** module options */
+{
+  jerry_value_t module = jerry_parse ((const jerry_char_t *) "", 0, (&module_parse_options));
```
Comment (zherczeg) [inline tests/unit-core/test-api-objecttype.c:46]: `(&module_parse_options)` -> `&module_parse_options`

File: tests/unit-core/test-api-objecttype.c [hunk 10]
```diff
@@ -40,6 +40,17 @@ test_ext_function (const jerry_call_info_t *call_info_p, /**< call information *
...
+  jerry_value_t modulelinked = jerry_module_link (module, NULL, NULL);
```
Comment (zherczeg) [inline tests/unit-core/test-api-objecttype.c:47]: module_linked

File: tests/unit-core/test-api-objecttype.c [hunk 11]
```diff
@@ -74,6 +85,11 @@ main (void)
   const jerry_char_t regexp_object[] = "new RegExp()";
   const jerry_char_t string_object[] = "new String('foo')";
   const jerry_char_t weak_ref_object[] = "new WeakRef({})";
+  const jerry_char_t error_object[] = "new Error()";
+
+  jerry_value_t arraybuffer = jerry_create_arraybuffer (10);
+
+  jerry_value_t dataview = jerry_create_dataview (arraybuffer, 0, 4);
```
Comment (zherczeg) [inline tests/unit-core/test-api-objecttype.c:92]: Move this to a helper function (similar to test_namespace)

**Review by zherczeg — CHANGES_REQUESTED** [review]

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by rerobika — APPROVED** [review]

**Pull request merged**